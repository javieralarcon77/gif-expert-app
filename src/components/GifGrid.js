import React from 'react'
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:imgs , loading } = useFetchGifs(category);
    
    return (
        <>
            <h3 className=" animate__animated animate__fadeIn">{ category }</h3>
            {
                loading &&
                <p className=" animate__animated animate__flash">Cargando...</p>
            }
            <div className="card-grid">
                {
                    imgs.map(img=>
                        <GifGridItem 
                            key={ img.id } 
                            { ...img } 
                        />
                    )
                }
            </div>
        </>
    )
}

GifGrid.protoTypes = {
    category: PropTypes.string.isRequired
}