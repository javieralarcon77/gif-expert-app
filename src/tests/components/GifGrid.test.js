import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

//pruebas usando mocks para simular la ejecucion de hook
describe('pruebas sobre el componente <GifGrid />',()=>{
    const category = 'One Piece'
    
    test('debe de verse correctamente el componente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs',()=>{
        const gifs = [{
            id: '1',
            title: 'Imagen 1',
            url:'https://localhost/img1.png'
        },
        {
            id: '2',
            title: 'Imagen 2',
            url:'https://localhost/img2.png'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={ category } />);
        
        //valida que la foto sea igual al anterior
        expect( wrapper ).toMatchSnapshot();

        //valida que el parrafo que muestra loading no exista
        expect( wrapper.find('p').exists() ).toBe( false );

        //valida que se creen los elmentos correctamente
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        

    });

})