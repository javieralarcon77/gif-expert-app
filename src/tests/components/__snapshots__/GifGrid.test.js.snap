// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="Imagen 1"
      url="https://localhost/img1.png"
    />
    <GifGridItem
      id="2"
      key="2"
      title="Imagen 2"
      url="https://localhost/img2.png"
    />
  </div>
</Fragment>
`;

exports[`pruebas sobre el componente <GifGrid /> debe de verse correctamente el componente 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
